### PR TITLE
Add 'show inconsistent code' arg to jadx

### PR DESCRIPTION
**Pull request by psolyca**
Add option to 'show bad code' during decompilation in the drop-down menu when opening an APK.

This option avoid "throw new UnsupportedOperationException("Method not decompiled: blablabla"
Code is not clean but it is better than nothing to RE.

### CODE REVIEW

**Review by Surendrajat — CHANGES_REQUESTED** [review]
@psolyca Thanks for the PR. I appreciate it.
However, due to some ugly hack we used earlier(L:54 & L:58), this option will fail to work in case the user does not select the `--deobf (jadx)` option. Could you please fix that?

**Comment by psolyca** [comment]
Ho sorry, I didn't see that as I use both options.
Will have a look.

**Comment by psolyca** [comment]
@Surendrajat Is this ok for you this way ?
It is maybe more "complexe" but more open to any changes later in `quick-pick` menu. No hardcoded values.

File: src/utils/quick-picks.ts [hunk 1]
```diff
@@ -45,38 +45,51 @@ const quickPickItems: { [index: string]: QuickPickItem[] } = {
...
-            label: "--deobf (jadx)",
-            detail: "Activate deobfuscation for Jadx",
+            label: "--deobf",
+            detail: "Activate deobfuscation",
+            description: "jadx",
+            alwaysShow: true,
+        },
+        {
+            label: "--show-bad-code",
+            detail: "show inconsistent code (incorrectly decompiled)",
```
Comment (Surendrajat) [inline src/utils/quick-picks.ts:91]:
```suggestion
            detail: "Show inconsistent code (incorrectly decompiled)",
```

File: src/interface.ts [hunk 2]
```diff
@@ -44,19 +41,29 @@ export namespace UI {
...
                 "Additional features & Apktool/Jadx arguments"
             );
-            if (args) {
-                const decompileJavaIndex = args.indexOf("decompile_java");
-                const quarkAnalysisIndex = args.indexOf("quark_analysis");
-                const jadxOptionsIndex = args.indexOf("--deobf");
+
+            if (quickPickItems) {
+                const args = quickPickItems.map<string>((item) => item.label);
+                const argsIndex = quickPickItems.map<string | undefined>(
```
Comment (Surendrajat) [inline src/interface.ts:51]: Index might reflect the intent but not super clear right away.
```suggestion
                const argDescriptions = quickPickItems.map<string | undefined>(
```

**Review by Surendrajat — CHANGES_REQUESTED** [review]
@psolyca Looks much better and works too. Thanks.
I've few small comments if you can address then we can merge it.

File: src/interface.ts [hunk 3]
```diff
@@ -44,19 +41,29 @@ export namespace UI {
...
+                    (item) => item.description
+                );
+                const decompileJavaIndex = argsIndex.indexOf("[Use Jadx]");
+                const quarkAnalysisIndex = argsIndex.indexOf(
+                    "[Use Quark-Engine]"
+                );
+                const jadxOptionsIndex = argsIndex.indexOf("jadx");
+                const jadxOptionsNumber = argsIndex.filter(
```
Comment (psolyca) [inline src/interface.ts:59]:
```suggestion
                const decompileJavaIndex = argDescriptions.indexOf("[Use Jadx]");
                const quarkAnalysisIndex = argDescriptions.indexOf(
                    "[Use Quark-Engine]"
                );
                const jadxOptionsIndex = argDescriptions.indexOf("jadx");
                const jadxOptionsNumber = argDescriptions.filter(
```
For the record

**Review by Surendrajat — APPROVED** [review]

**Pull request merged**